packages/smooth_app/lib/helpers/product_compatibility_helper.dart

```diff
@@ -1,137 +1,45 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
-import 'package:openfoodfacts/model/Attribute.dart';
-import 'package:openfoodfacts/model/AttributeGroup.dart';
-import 'package:openfoodfacts/model/Product.dart';
-import 'package:openfoodfacts/personalized_search/preference_importance.dart';
-import 'package:smooth_app/data_models/product_preferences.dart';
-import 'package:smooth_app/helpers/attributes_card_helper.dart';
-
-enum ProductCompatibility {
-  UNKNOWN,
-  INCOMPATIBLE,
-  BAD_COMPATIBILITY,
-  NEUTRAL_COMPATIBILITY,
-  GOOD_COMPATIBILITY,
-}
-
-class ProductCompatibilityResult {
-  ProductCompatibilityResult(
-      this.averageAttributeMatch, this.productCompatibility);
-  final num averageAttributeMatch;
-  final ProductCompatibility productCompatibility;
-}
-
-const int _BAD_COMPATIBILITY_UPPER_THRESHOLD = 33;
-const int _NEUTRAL_COMPATIBILITY_UPPER_THRESHOLD = 66;
-
-// Defines the weight of an attribute while computing the average match score
-// for the product. The weight depends upon it's importance set in user prefs.
-const Map<String, int> attributeImportanceWeight = <String, int>{
```

Comment:
so this was an algorithm written by Stephane to compute the product compatibility score, does `MatchedProduct` adhere to the same algorithm ? I'm not sure, but it doesn't seem like it, and it should :)

Reply:
What we can agree on is that we should use only one algorithm.
@teolemon seemed to say that 5 levels were considered by end-users as too confusing and that 3 are enough.
And my algorithm too was initially written by Stéphane, I just converted it to dart.
Which algorithm should we keep? No idea.

Reply:
@stephanegigandet

Reply:
@stephanegigandet @teolemon @jasmeet0817 @M123-dev My opinion: the main point here is to have a unique score algorithm on the app (much better for end-users, UX-wise), and that is implemented by this PR.

We can later say: maybe we should improve the score algorithm, add a possible enum value, fix here and there.
But for the moment, that looks good enough.